brave/brave-ios
# Fix #7370: NFT Discovery

## Summary of Changes
NFT discovery can be enabled via the Auto discovery dialog or via the Enable NFT Discovery in settings under Web3.


This pull request fixes #7370

## Submitter Checklist:


## Test Plan:

Case 1: Check if `Enable NFT Discovery` is available for user to configure under Web3 settings
1. open Brave. Make a wallet account if you dont have one
2. go to the web3 settings screen from wallet or from the browser settings
3. Check if `Enable NFT Discovery` is available for user to config. If you have never configure this preference. The default value should be false (toggle off)

Case 2: The NFT discovery dialog
1. open Brave. Make a wallet account if you dont have one
2. go to the web3 settings screen from wallet or from the browser settings
3. Turn on `Enable NFT Discovery`
4. Go to NFT tab
5. There should be not NFT discovery dialog being prompt 
6. Go back to web3 settings and turn off `Enable NFT Discovery`
7. Go back to NFT tab
8. You should see a NFT discovery dialog appears

Case 3: The NFT discovery dialog should show open [Learn more](https://github.com/brave/brave-browser/wiki/NFT-Discovery) link.
1. From Case 2 step 8, check if `Learn more` will bring you to the correct website.

Case 4: The NFT discovery dialog should never be shown after selecting either option.
1. from case 3 step 1, go back to Wallet and NFT tab
2. the NFT discovery dialog should appear again since no decision has been made
3. Click either yes or no button
4. The dialog will be dismissed 
5. if you click yes, stay inside Wallet or exit wallet. but need 1 or 2 min gap to kill the browser app and relaunch again
6. Unlock into the wallet and go to NFT Tab
7. all the NFTs that you wallet account owns will appear in the tab
8. Once you have made decision on either enable or disable NFT discovery 
9. Check `Enable NFT Discovery` under web3 display the correct preference
10. And there won't be NFT discovery dialog showing up if you go to NFT tab. 

Case 5: Should able to import new NFT if user has no visible NFT.
1. open Brave, Make a wallet account if you dont have one
2. make sure you dont own any NFT or you choose not to auto-discover NFT
3. there will be an import button in NFT tab
4. Click import button
5. A form will show up for you to input NFT information so it can be added as a custom asset
6. Input required info and click Add to add the NFT
7. Newly added NFT should appear in the NFT tab.


## Screenshots:



https://user-images.githubusercontent.com/1187676/236557019-d5a79797-f30e-4878-bb89-4e98f4a5343a.mov

https://user-images.githubusercontent.com/1187676/236557048-a83abdfb-fe70-4152-9a5a-5b5383d73177.mov


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
